planet/cli/orders.py

```diff
@@ -221,6 +221,35 @@ async def create(ctx, request: str, pretty):
     echo_json(order, pretty)
 
 
+def stash_item_type(ctx, param, value):
+    ctx.obj['item_type'] = value
+    return value
+
+
+def bundle_cb(ctx, param, value):
+    item_type = ctx.obj['item_type']
+    choices = planet.specs.get_product_bundles(item_type)
+    if value is None:
+        msg = ("Choose from:\n\t{choices}").format(
+            choices=",\n\t".join(choices))
+        raise click.ClickException(f"Missing option '--{param.name}'. {msg}")
+    else:
+        return value
+
+
+class Bundle(click.Choice):
+    name = "bundle"
+
+    def __init__(self, case_sensitive: bool = True) -> None:
+        self.choices = []
+        self.case_sensitive = case_sensitive
+
+    def convert(self, value, param, ctx):
+        item_type = ctx.obj['item_type']
+        self.choices = planet.specs.get_product_bundles(item_type)
+        return super().convert(value, param, ctx)
```

Comment:
@kevinlacaille this class and the functions above it need some docstrings and comments to explain their purpose. Like I said yesterday, this is a bit hacky, and I'm concerned that we might eventually forget why we took this particular approach or how it works.